src/d3a/models/read_user_profile.py

```diff
@@ -289,12 +304,17 @@ def read_arbitrary_profile(profile_type: InputProfileTypes,
     or a dict with hourly data (Dict[int, float])
     or a dict with arbitrary time data (Dict[str, float])
     or a string containing a serialized dict of the aforementioned structure
+    :param copy:
     :return: a mapping from time to profile values
     """
 
     profile = _read_from_different_sources_todict(input_profile)
+    profile_time_list = list(profile.keys())
+    profile_duration = profile_time_list[-1] - profile_time_list[0]
+    if GlobalConfig.sim_duration > duration(days=1) and profile_duration <= duration(days=1):
```

Comment:
I can understand the reason for the less than or equal (instead of just equal) in the last block, but I am afraid it is going to bite us in the future. Can you please explain what happens if the profile_duration is less than a day ? Because I do not recall what happens in that occasion.

Reply:
If the profile is provided only e.g. until 12 o'clock, the last value will be used for the rest of the day (12:00-23:45). 
If a user provides profile less than a day, this behaviour should also be applied to the following days. (IMO, what do you think?)